### ISSUE TITLE
Rename SbieDll

### ISSUE THREAD

**Issue by H4CK3R0FF1C14L** [sandboxie-plus/Sandboxie#3527]
### Is your feature request related to a problem or use case?

How can I rename SbieDll to avoid detection by the program? And since I can generally build this project, there are no build instructions and it makes it very difficult to work with this project?

### Describe the solution you'd like

 .

**Comment by isaak654** [comment]
> How can I rename SbieDll to avoid detection by the program?

You can find references and cross-references about it in #2880

> And since I can generally build this project, there are no build instructions and it makes it very difficult to work with this project?

Sandboxie Classic only:
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Classic

Sandboxie Plus only:
https://github.com/sandboxie-plus/Sandboxie/discussions/1089#discussioncomment-2441183
https://github.com/sandboxie-plus/Sandboxie/tree/master/Installer
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Plus

If the request is about adding additional documentation content in the repository files, please provide an explanation or a pull request of how this should be done from your perspective.

**Comment by isaak654** [comment]
> How can I rename SbieDll to avoid detection by the program?

You can find references and cross-references about it in #2880

> And since I can generally build this project, there are no build instructions and it makes it very difficult to work with this project?

Sandboxie Classic only:
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Classic

Sandboxie Plus only:
https://github.com/sandboxie-plus/Sandboxie/discussions/1089#discussioncomment-2441183
https://github.com/sandboxie-plus/Sandboxie/tree/master/Installer
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Plus

If you're interested about adding further documentation content in the repository files, please provide an explanation or a pull request of how this should be done from your perspective.

**Comment by isaak654** [comment]
> How can I rename SbieDll to avoid detection by the program?

You can find references and cross-references about it in #2880

> And since I can generally build this project, there are no build instructions and it makes it very difficult to work with this project?

Sandboxie Classic:
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22build+issue%22
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Classic

Sandboxie Plus:
https://github.com/sandboxie-plus/Sandboxie/discussions/1089#discussioncomment-2441183
https://github.com/sandboxie-plus/Sandboxie/tree/master/Installer
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22build+issue%22
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Plus

If you're interested about adding further documentation content in the repository files, please provide an explanation or a pull request of how this should be done from your perspective.

**Comment by isaak654** [comment]
> How can I rename SbieDll to avoid detection by the program?

You can find references and cross-references about it in #2880

> And since I can generally build this project, there are no build instructions and it makes it very difficult to work with this project?

Sandboxie Classic:
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Classic

Sandboxie Plus:
https://github.com/sandboxie-plus/Sandboxie/discussions/1089#discussioncomment-2441183
https://github.com/sandboxie-plus/Sandboxie/tree/master/Installer
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Plus

Documentation category, build issue label, documentation label:
https://github.com/sandboxie-plus/Sandboxie/discussions/categories/documentation-and-notes
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22build+issue%22
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22documentation%22

If you're interested about adding further documentation content in the repository files, please provide an explanation or a pull request of how this should be done from your perspective.

**Comment by isaak654** [comment]
> How can I rename SbieDll to avoid detection by the program?

You can find references and cross-references about it in #2880

> And since I can generally build this project, there are no build instructions and it makes it very difficult to work with this project?

Sandboxie Classic:
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Classic

Sandboxie Plus:
https://github.com/sandboxie-plus/Sandboxie/discussions/1089#discussioncomment-2441183
https://github.com/sandboxie-plus/Sandboxie/tree/master/Installer
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Plus

Documentation category, build issue label, documentation label, how to label:
https://github.com/sandboxie-plus/Sandboxie/discussions/categories/documentation-and-notes
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22build+issue%22
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22documentation%22
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22how+to%22

If you're interested about adding further documentation content in the repository files, please provide an explanation or a pull request of how this should be done from your perspective.

**Comment by isaak654** [comment]
> How can I rename SbieDll to avoid detection by the program?

You can find references and cross-references about it in #2880

> And since I can generally build this project, there are no build instructions and it makes it very difficult to work with this project?

- Build instructions and content related to Sandboxie Classic:
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/Sandboxie/install/ReadMe.md
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Classic

- Build instructions and content related to Sandboxie Plus:
https://docs.google.com/document/d/1iUZVUnfwpmGdj_m6_MZ3SVn4756jOk14FQ0sfek0TTI
https://github.com/sandboxie-plus/Sandboxie/tree/master/Installer
https://github.com/sandboxie-plus/Sandboxie/blob/master/LICENSE.Plus

- Documentation category and related labels:
https://github.com/sandboxie-plus/Sandboxie/discussions/categories/documentation-and-notes
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22build+issue%22
https://github.com/sandboxie-plus/Sandboxie/issues?q=label%3A%22documentation%22

- https://github.com/sandboxie-plus/Sandboxie/pull/3092

If you're interested about adding further documentation content in the repository files, please provide a detailed explanation or a pull request of how this should be done from your perspective.

**Comment by H4CK3R0FF1C14L** [comment]
> * Build instructions and content related to Sandboxie Plus:
>   https://docs.google.com/document/d/1iUZVUnfwpmGdj_m6_MZ3SVn4756jOk14FQ0sfek0TTI

There are no problems with building Sandboxie itself, but Sandboxie-Plus gives me QT-related errors. I first installed QT 5.12.x which is specified in this document, but found that there is no msvc 2019, other sources specify version 5.15, as well as in the yaml file from Github Actions also specified version 5.15, I installed it, downloaded the plugin for VS 2019, added, but still get errors related to QT but different.

```
Error	C1083	Cannot open include file: 'QtWin': No such file or directory SandMan 
\Sandboxie\SandboxiePlus\SandMan\Helpers\WinHelper.cpp	
```

Also in the Sandboxie-Plus folder there are cmd files that are used in the GitHub Actions yaml file and I don't really understand what they are for, as running install_jom.cmd I obviously don't do what I should, I get the QT folder on the desktop, then running qmake_plus.cmd it gives errors related to the fact that it can't find some QT files (most likely the ones I have installed on my C drive, but it looks for them inside the same folder). I would like to know more about the Sandboxie-Plus build, as it causes a lot of problems for me.

**Comment by isaak654** [comment]
There are several typos that should be fixed in that document, unfortunately I don't have the permission to do it from there. If someone would have the courtesy to use it as a basis for a new pull request to create a new [ReadMe.md](https://github.com/sandboxie-plus/Sandboxie/tree/master/SandboxiePlus) file, I would be glad to supervise and merge it.

**Comment by H4CK3R0FF1C14L** [comment]
I installed QT version 6.3.x and was able to build Sandboxie-Plus under x64, but there is no msvc2019 under x32 which makes it impossible for me to build the x32 version. And as far as I understand using QT 6 version Sandboxie-Plus will not work on older Windows versions.

```
5>SandMan.vcxproj -> C:\Users\donet\Desktop\Sandboxie\SandboxiePlus\x64\Release\SandMan.exe
5>Done building project "SandMan.vcxproj".
========== Build: 5 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
```

**Comment by offhub** [comment]
> version 5.15, I installed it, downloaded the plugin for VS 2019, added, but still get errors related to QT but different.
> > 
> > ```
> > Error	C1083	Cannot open include file: 'QtWin': No such file or directory SandMan 
> > \Sandboxie\SandboxiePlus\SandMan\Helpers\WinHelper.cpp	
> > ```

For Qt 5.15
Visual Studio 2022 -> SandMan -> Properties -> Configuration Properties ->
Qt Project Settings -> QtModules -> Click the down arrow next to it to open the `Select Modules` window ->
Select the `WindowExtras` module from the window that opens > OK

> [!NOTE]
> Menus may differ in the 2019 version.

**Comment by isaak654** [comment]
> version 5.15, I installed it, downloaded the plugin for VS 2019, added, but still get errors related to QT but different.
>
> Error	C1083	Cannot open include file: 'QtWin': No such file or directory SandMan 
> \Sandboxie\SandboxiePlus\SandMan\Helpers\WinHelper.cpp

You can also try with commit https://github.com/sandboxie-plus/Sandboxie/pull/3092/commits/251d7cc0fd8aad62a0c81d24ae6521b1c83ec8ed

I'm not sure why winextras was not included there, is a new pull request needed?

**Comment by isaak654** [comment]
> version 5.15, I installed it, downloaded the plugin for VS 2019, added, but still get errors related to QT but different.
> 
> ```
> Error	C1083	Cannot open include file: 'QtWin': No such file or directory SandMan 
> \Sandboxie\SandboxiePlus\SandMan\Helpers\WinHelper.cpp	
> ```


You can also try with commit https://github.com/sandboxie-plus/Sandboxie/pull/3092/commits/251d7cc0fd8aad62a0c81d24ae6521b1c83ec8ed

I'm not sure why winextras was not included there, is a new pull request needed?

**Comment by isaak654** [comment]
> version 5.15, I installed it, downloaded the plugin for VS 2019, added, but still get errors related to QT but different.
> 
> ```
> Error	C1083	Cannot open include file: 'QtWin': No such file or directory SandMan 
> \Sandboxie\SandboxiePlus\SandMan\Helpers\WinHelper.cpp	
> ```


You can also try with commit https://github.com/sandboxie-plus/Sandboxie/pull/3092/commits/251d7cc0fd8aad62a0c81d24ae6521b1c83ec8ed

I'm not sure why winextras was not included in master, is a new pull request needed?

**Comment by H4CK3R0FF1C14L** [comment]
Yes, this fixes an issue with a build using QT version 5.15.x. Now it remains to figure out how to rename the SbieDll library to a different name. 

> You can also try with commit [251d7cc](https://github.com/sandboxie-plus/Sandboxie/commit/251d7cc0fd8aad62a0c81d24ae6521b1c83ec8ed)

**Comment by H4CK3R0FF1C14L** [comment]
I have renamed all .vcproj, .h, .rc, .cpp files wherever I can find "SbieDll.dll" and "SbieDll.lib" but I seem to be missing something. The application is fully built via Github Actions (except for the ARM version) but I get this error when I try to run anything through the sandbox.

![image](https://github.com/sandboxie-plus/Sandboxie/assets/133565745/4a6d3925-ca62-49f9-8877-57466f28083c)

**Comment by isaak654** [comment]
There is an existing troubleshooting wizard for SBIE2181, which can be useful for non-standard Sandboxie installation paths: 

- https://github.com/sandboxie-plus/Sandboxie/issues/3503#issuecomment-1861237653

**Comment by H4CK3R0FF1C14L** [comment]
> * [All browsers do not work in sandboxie #3503 (comment)](https://github.com/sandboxie-plus/Sandboxie/issues/3503#issuecomment-1861237653)

This doesn't solve the problem, when I try to launch anything, I have Start.exe appear for a second in my box and disappear after a second, not even the application selection window appears.

![image](https://github.com/sandboxie-plus/Sandboxie/assets/133565745/63586d2b-4c2f-4039-9208-ca58a1234374)

**Comment by isaak654** [comment]
What is your Windows version?

**Comment by H4CK3R0FF1C14L** [comment]
> What is your Windows version?

Windows 11 Pro 22H2

**Comment by H4CK3R0FF1C14L** [comment]
It seems that I'm just an idiot who accidentally using CTRL + H in the code editor, replaced also the name of the file sbiedll.h because of which nothing worked (it's strange that it was built under x86/64 but under ARM it gave an error, which I didn't care about because I don't own ARM devices). Also I forgot to change the field in my_headers.h (although actually I remember that I changed it, but apparently I forgot to save it).

ARM version still does not build because it is missing some make file, but it does not matter in general. XD

**Comment by isaak654** [comment]
> ARM version still does not build because it is missing some make file

This has recently been fixed with commit f8377e14d9ca6295199ef7592ea9602275871c36
